src/mpyl/steps/deploy/k8s/rancher.py

```diff
@@ -2,32 +2,40 @@
 
 from dataclasses import dataclass
 
+from ....steps import Input
 from ....project import Target
 
 
-@dataclass(frozen=True)
+@dataclass()
 class ClusterConfig:
     project_id: str
     cluster_id: str
     cluster_env: str
     context: str
 
+    def __init__(self, config: dict):
```

Comment:
No need to create an __init__ in a dataclass

Reply:
Yeah I wasn't sure about this part, how would you apply the logic that's currently in the `init` then? This is more clean than having to pass `cluster_configs['test'].projectId, cluster_configs['test'].clusterId, etc` for all 3 environments.

Reply:
hmm yeah fair enough.. I think in other places we do this as well, and call the method `from_config()` or so

Reply:
And it can be frozen as well